### PR TITLE
sync_music: init at 0.5.0

### PR DESCRIPTION
**Pull request by gaykitty**
## Description of changes

https://github.com/fetzerch/sync_music

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by WolfangAukang** [comment]
- Is it possible to build it from source? There are tests in the source repo.
- Also, [please add new programs to `pkgs/by-name`](https://github.com/NixOS/nixpkgs/tree/master/pkgs#quick-start-to-adding-a-package).

**Comment by gaykitty** [comment]
I think that should do it?

File: pkgs/by-name/sy/sync_music/package.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+, python3Packages
+, fetchFromGitHub
+, ffmpeg
+, mutagen
+}:
+
+python3Packages.buildPythonPackage rec {
+  pname = "sync_music";
+  version = "0.5.0";
+  format = "setuptools";
```
Comment (WolfangAukang) [inline pkgs/by-name/sy/sync_music/package.nix:11]: https://github.com/NixOS/nixpkgs/pull/254136

```suggestion
  pyproject = true;
```

File: pkgs/by-name/sy/sync_music/package.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    owner = "fetzerch";
+    repo = "sync_music";
+    rev = "v${version}";
+    hash = "sha256-8vDSTaTQvdADTSxbfSxgCDsFbGPD5RtQGURv3LTwGXQ=";
+  };
+
+  propagatedBuildInputs = [
+    python3Packages.mutagen
+    python3Packages.pbr
+    ffmpeg
```
Comment (WolfangAukang) [inline pkgs/by-name/sy/sync_music/package.nix:23]: Related to the `pyproject` recommendation above, to make it build with setuptools.
```suggestion
    python3Packages.setuptools
    ffmpeg
```

File: pkgs/by-name/sy/sync_music/package.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  # Manually set version because prb wants to get it from the git
+  # upstream repository (and we are installing from tarball instead)
+  env.PBR_VERSION = version;
+
+  setuptoolsCheckPhase = "true";
+
+  # checkPhase = ''
+  #   runHook preCheck
+  #   runHook postCheck
+  # '';
```
Comment (WolfangAukang) [inline pkgs/by-name/sy/sync_music/package.nix:42]: if the checks work, you can remove this

File: pkgs/by-name/sy/sync_music/package.nix [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  meta = with lib; {
+    description = " Sync music library to external devices ";
```
Comment (WolfangAukang) [inline pkgs/by-name/sy/sync_music/package.nix:45]:
```suggestion
    description = " Sync music library to external devices";
    homepage = "https://github.com/fetzerch/sync_music";
    changelog = "https://github.com/fetzerch/sync_music/releases/tag/${src.rev}";
    mainProgram = "sync_music";
```

**Review by WolfangAukang — CHANGES_REQUESTED** [review]

**Comment by WolfangAukang** [comment]
One final thing, it is preferred that the package name is hyphenated, according to the recommendation stated [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming):

> Dashes in the package pname should be preserved in new variable names, rather than converted to underscores or camel cased — e.g., http-parser instead of http_parser or httpParser. The hyphenated style is preferred in all three package names.

**Comment by gaykitty** [comment]
Changes applied, thx

**Review by WolfangAukang — APPROVED** [review]

**Comment by WolfangAukang** [comment]
Result of `nixpkgs-review pr 276515` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>sync-music</li>
    <li>sync-music.dist</li>
  </ul>
</details>

**Review by WolfangAukang — APPROVED** [review]

**Comment by gaykitty** [comment]
I found a bug and added a patch for it here as well as submitting it upstream. https://github.com/fetzerch/sync_music/pull/18

**Comment by gaykitty** [comment]
I'm going to close this PR for now. While I do still use this program personally, it appears unmaintained so it may not be suitable for nixpkgs.

**Pull request closed without merging**